[IT] Added `Client/File_Formats/*` (#9701)
ppy/osu-wiki: wiki/Client/File_formats/osu_(file_format)/it.md

# .osu (formato di file)

**`.osu`** è un formato di file leggibile dall'uomo contenente informazioni su una beatmap.

## Struttura

La prima riga del file specifica la versione del formato del file. Ad esempio, `osu file format v14` è la versione più recente.

Il contenuto seguente è separato in sezioni, indicate dai titoli delle sezioni tra parentesi quadre.

> | Sezione | Descrizione | Tipo di contenuto

Reviewer: ```suggestion
| Sezione | Descrizione | Tipo di contenuto |
```

| :-- | :-- | :-- |
| `[Generale]` | Informazioni generali sulla beatmap | Coppie `chiave: valore' |
| `[Editor]` | Impostazioni salvate per l'editor della beatmap | `chiave: valore` coppie |
| `[Metadata]` | [Informazioni](/wiki/Client/Beatmap_editor/Song_setup#song-and-map-metadata) utilizzate per identificare la beatmap | `chiave:valore` coppie |
| `[Difficoltà]` | [Impostazioni di difficoltà](/wiki/Client/Beatmap_editor/Song_setup#difficulty) | coppie `chiave:valore` |
| `[Eventi]` | Eventi grafici di beatmap e storyboard | Elenchi separati da virgole |
| `[TimingPoints]` | Punti di cronometraggio e controllo | Elenchi separati da virgole |
| `[Colours]` | Colori della combo e della skin | Coppie `key : value` |
| `[HitObjects]` Oggetti colpiti | Elenchi separati da virgole |

## Generale

<!-- TODO: this is missing some functional options that are leftover from very old file formats -->

> | Opzione | Tipo di valore | Descrizione | Valore predefinito

Reviewer: ```suggestion
| Opzione | Tipo di valore | Descrizione | Valore predefinito |
```

| :-- | :-- | :-- | :-- |
| `AudioFilename` | Stringa | Posizione del file audio rispetto alla cartella corrente | |
| `AudioLeadIn` | Integer | Millisecondi di silenzio prima dell'inizio della riproduzione dell'audio | 0 |
| `AudioHash` | Stringa | *Deprecated* | |
| `PreviewTime` | Integer | Tempo in millisecondi in cui deve iniziare l'anteprima audio | -1 |
| `Countdown` | Integer | Velocità del conto alla rovescia prima del primo oggetto colpito (`0` = nessun conto alla rovescia, `1` = normale, `2` = metà, `3` = doppio) | 1
| `SampleSet` | Stringa | Set di campioni che verrà utilizzato se i punti di temporizzazione non lo sovrascrivono (`Normal`, `Soft`, `Drum`) | Normal |
| `StackLeniency` | Decimale | [Moltiplicatore](/wiki/Beatmap/Stack_leniency) per la soglia nel tempo in cui gli oggetti colpiti posti vicini si impilano (0-1) | 0,7 |

> | `Mode` | Integer | Game mode (`0` = osu!, `1` = osu!taiko, `2` = osu!catch, `3` = osu!mania) | 0

Reviewer: ```suggestion
| `Mode` | Integer | Game mode (`0` = osu!, `1` = osu!taiko, `2` = osu!catch, `3` = osu!mania) | 0 |
```

| `LetterboxInBreaks` | 0 o 1 | Se le interruzioni hanno o meno un effetto letterboxing | 0 |
| `StoryFireInFront` | 0 o 1 | *Deprecated* | 1 |
| `UseSkinSprites` | 0 o 1 | Se lo storyboard può usare o meno le immagini della skin dell'utente | 0 |
| `AlwaysShowPlayfield` | 0 o 1 | *Deprecated* | 0 |

> | `OverlayPosition` | Stringa | Sorteggia l'ordine di sovrapposizione dei cerchi colpiti rispetto ai numeri colpiti (`NoChange` = usa l'impostazione della skin, `Below` = sorteggia le sovrapposizioni sotto i numeri, `Above` = sorteggia le sovrapposizioni sopra i numeri) | NoChange

Reviewer: ```suggestion
| `OverlayPosition` | Stringa | Sorteggia l'ordine di sovrapposizione dei cerchi colpiti rispetto ai numeri colpiti (`NoChange` = usa l'impostazione della skin, `Below` = sorteggia le sovrapposizioni sotto i numeri, `Above` = sorteggia le sovrapposizioni sopra i numeri) | NoChange |
```

| `WidescreenStoryboard` | 0 o 1 | Se lo storyboard consente o meno la visualizzazione in widescreen | 0 |
| `SamplesMatchPlaybackRate` | 0 o 1 | Se i campioni sonori cambieranno o meno velocità quando si gioca con mod che cambiano velocità | 0 |

## Editor

Queste opzioni sono rilevanti solo quando si aprono le mappe nell'[editor delle beatmap](/wiki/Client/Beatmap_editor). Non influiscono sulla giocabilità.

| Opzione | Tipo di valore | Descrizione |
| :-- | :-- | :-- |
| `Bookmarks` | Elenco separato da virgole di numeri interi | Tempo in millisecondi dei [bookmarks](/wiki/Client/Beatmap_editor/Compose#bottom-(song's-timeline)) |
| `DistanceSpacing` | Decimale | Moltiplicatore dei [Distance snap](/wiki/Client/Beatmap_editor/Distance_snap) |
| `BeatDivisor` | Integer | [Beat snap divisor](/wiki/Client/Beatmap_editor/Beat_snap_divisor) |
| `GridSize` | Integer | [Dimensione griglia](/wiki/Beatmapping/Grid_snapping) |
| `TimelineZoom` | Decimale | Fattore di scala per la [timeline degli oggetti](/wiki/Client/Beatmap_editor/Compose#top-left-(hit-objects-timeline)) |

## Metadata

> | Opzione | Tipo di valore | Descrizione

Reviewer: ```suggestion
| Opzione | Tipo di valore | Descrizione |
```

| :-- | :-- | :-- |
| `Title` | Stringa | Titolo romanizzato del brano |
| `TitleUnicode` | Stringa | Titolo del brano |
| `Artista` | Stringa | Artista romanizzato della canzone |
| `ArtistUnicode` | Stringa | Artista della canzone |
| `Creator` | Stringa | Creatore della beatmap |
| `Versione` | Stringa | Nome della difficoltà |
| `Source` | Stringa | Media originale per cui il brano è stato prodotto |
| `Tags` | Elenco di stringhe separate dallo spazio | Termini di ricerca |
| `BeatmapID` | Integer | ID della difficoltà |
| `BeatmapSetID` | Integro | ID beatmap |

## Difficulty

> | Opzione | Tipo di valore | Descrizione

Reviewer: ```suggestion
| Opzione | Tipo di valore | Descrizione |
```

- **`x` (Integer)** e **`y` (Integer):** Posizione in [pixel](/wiki/Client/Beatmap_editor/osu!_pixel) dell'oggetto.
- **`time` (Integer):** Tempo in cui l'oggetto deve essere colpito, in millisecondi dall'inizio dell'audio della beatmap.
- **`type` (Integer):** Bit flag che indicano il tipo di oggetto da cliccare. Vedi [la sezione tipi](#tipi).
- **`hitSound` (Integer):** Bit flag che indica l'hitsound applicato all'oggetto. Si veda [la sezione hitsounds](#hitsounds).
- **`objectParams` (Elenco separato da virgole):** Parametri extra specifici del tipo di oggetto.
- **`hitSample` (elenco separato da due colonne):** Informazioni su quali campioni vengono riprodotti quando l'oggetto viene colpito. È strettamente correlato a `hitSound`; vedere [la sezione hitsounds](#hitsounds). Se non viene scritto, il valore predefinito è `0:0:0:0:`.

### Tipi

I tipi di oggetti da colpire sono memorizzati in un integer a 8 bit, dove ogni bit è un flag con un significato speciale. Il tipo di oggetto da colpire di base è dato dai bit 0, 1, 3 e 7 (dal meno significativo al più significativo):

- 0: Cerchio
- 1: Cursore
- 3: Spinner
- 7: osu!mania hold

> 

Reviewer: ```suggestion
| 2 | Segna l'inizio di una nuova combo |
```

> I bit rimanenti sono utilizzati per distinguere le nuove combo e per saltare i colori delle combo (comunemente chiamato "colour hax"):

Reviewer: ```suggestion
| 3 | Segna l'oggetto come spinner  |
```

> 

Reviewer: ```suggestion
| 4, 5, 6 | Un integer a 3 bit che specifica quanti colori della combo saltare, una pratica alla quale ci si riferisce con "colour hax". Rilevante solo se l'oggetto inizia una nuova combo |
```

> - 2: Nuova combo

Reviewer: ```suggestion
| 7 | Segna l'oggetto come nota lunga di osu!mania |
```

> - 4-6: un intero a 3 bit che specifica quanti colori della combo saltare, se questo oggetto inizia una nuova combo.

Reviewer: ```suggestion
```

### Hitsounds

I bit flag `hitSound` determinano quali suoni verranno riprodotti quando l'oggetto viene colpito:

> 

Reviewer: ```suggestion

| Bit index | Meaning |
| :-- | :-- |
```

> - 0: Normale

Reviewer: ```suggestion
| 0 | Normal |
```

> - 1: Fischio

Reviewer: ```suggestion
| 1 | Whistle |
```

> - 2: Fine

Reviewer: ```suggestion
| 2 | Finish |
```

> - 3: Battito di mani

Reviewer: ```suggestion
| 3 | Clap |
```

#### Gli hitsound dei cursori

Oltre agli hitsound dei bordi, i cursori hanno anche un hitsound continuo ogni volta che il giocatore si trova nel raggio d'azione del cerchio di inseguimento del cursore. Il file sonoro viene riprodotto in loop per tutto il tempo in cui è attivo.

Questo hitsound utilizza le proprietà `hitSound` e `hitSample` dell'oggetto hit, ma sono supportati solo i suoni normale e fischio. Il nome del file è `<sampleSet>-slider<hitSound><index>.wav`, dove `hitSound` è `slide` per i suoni normali o `whistle` per i suoni fischiati.

### Spinner

*Sintassi degli spinner:* `x,y,time,type,hitSound,endTime,hitSample`

- **`endTime` (intero):** Tempo di fine dello spinner, in millisecondi dall'inizio dell'audio della beatmap.
- `x` e `y` non influiscono sugli spinner. Sono predefiniti al centro del campo di gioco, `256,192`.

> ### Trattenute (solo osu!mania)

Reviewer: ```suggestion
### Note lunghe (solo osu!mania)
```

### osu!taiko

Gli oggetti hit di osu!taiko usano solo `time` per determinare come sono posizionati sul campo di gioco, quindi `x` e `y` sono ignorati. Allo stesso modo, l'unica parte significativa delle curve degli slider è la `lunghezza`; `curveType` e `curvePoints` sono rilevanti solo quando si apre la mappa nell'editor. I colori delle combo e le nuove combo sono ignorati e vengono utilizzati gli hitsound specifici della modalità.

- I cerchi colpiti con hitsound fischio o battito di mani diventano kat, mentre gli altri cerchi colpiti diventano don. L'aggiunta dell'hitsound finish li trasforma nelle loro varianti grandi.
- I cursori diventano rulli di tamburo.
- Gli spinner diventano note denden.

### osu!catch

Il campo di gioco di osu!catch utilizza solo l'asse x, quindi `y` non è rilevante. Le curve dei cursori possono utilizzare lo spazio verticale per ottenere un'accelerazione orizzontale quando vengono appiattite in un campo di gioco monodimensionale.

- I cerchi dei colpi diventano frutti.
- I cursori diventano flussi di succo, con frutti su ogni bordo.
- Gli spinner diventano docce di banane

### osu!mania

> Analogamente a osu!catch, gli oggetti colpiti in osu!mania non usano `y`. Per determinare la colonna si usa `x`; si veda la sezione [Holds] (#holds-(osu!mania-only)).

Reviewer: ```suggestion
Analogamente a osu!catch, gli oggetti colpiti in osu!mania non usano `y`. Per determinare la colonna si usa `x`; si veda la sezione [Note lunghe] (#note-lunghe-(solo-osu!mania)).
```

> Analogamente a osu!catch, gli oggetti colpiti in osu!mania non usano `y`. Per determinare la colonna si usa `x`; si veda la sezione [Note lunghe] (#note-lunghe-(solo-osu!mania)).

Reviewer: ```suggestion
Analogamente a osu!catch, gli oggetti colpiti in osu!mania non usano `y`. Per determinare la colonna si usa `x`; si veda la sezione [Note lunghe](#note-lunghe-(solo-osu!mania)).
```
